syntax = "proto3";
package pb;

message actionRequest {
  string text = 1;
  enum Corpus {
    UNIVERSAL = 0;
    WEB = 1;
    IMAGES = 2;
    LOCAL = 3;
    NEWS = 4;
    PRODUCTS = 5;
    VIDEO = 6;
  }
  Corpus corpus = 4;
}
message actionResponse{
  string text = 1;
  int32 age = 2;
  message Result {
    string url = 1;
    string title = 2;
    repeated string snippets = 3;
  }
  repeated Result result = 3;
}

service FormatData {
  rpc DoFormat(actionRequest) returns (actionResponse){}
}